libraries/stdlib/samples/test/samples/text/strings.kt

```diff
@@ -79,4 +79,196 @@ class Strings {
         assertPrints(byteArray.toString(charset), "Hello")
     }
 
+
+    //Indices is IntRange
+    @Sample
+    fun slice() {
+        assertPrints("abcdef".slice(IntRange(1,2)),"bc")
+        assertPrints("abcdef".slice(IntRange(0,0)),"a")
+    }
+
+    //Without the startIndex
+    @Sample
+    fun startsWith(){
+        assertPrints("abcd".startsWith("ab",true),"true")
+        assertPrints("abcd".startsWith("Ab",true),"true")
+        assertPrints("Abcd".startsWith("Ab",false),"true")
+        assertPrints("abcd".startsWith("ab",false),"true")
+
+    }
+
+    //With the startIndex
+    @Sample
+    fun startsWithStartIndex(){
+        assertPrints("abcd".startsWith("B",1,true),"true")
+        assertPrints("abcd".startsWith("c",0,true),"false")
+        assertPrints("abcd".startsWith("Ab",0,true),"true")
+        assertPrints("Abcd".startsWith("Ab",0,false),"true")
+        assertPrints("abcd".startsWith("c",2,false),"true")
+        assertPrints("abcd".startsWith("C",2,false),"false")
+    }
+
+    //delimiter: Char
+
+    @Sample
+    fun substringBefore(){
+        assertPrints("abcd".substringBefore("c"),"ab")
+        assertPrints("abcd".substringBefore("8"),"abcd")
+        assertPrints("kotlinot".substringBefore("ot"),"k")
+        assertPrints("kotlin".substringBefore('k'),"")
```

Comment:
I think it'd be ok to merge first and third sample. 
Better to use some real use case to illustrate these functions, for example to take a part of a dot-delimitied identifier, or a slash-delimited path.

Reply:
Could you also show the usage of these functions with `missingDelimiterValue` parameter specified?